### ISSUE TITLE
ObjectIds are distinct from document _id fields

### ISSUE THREAD

**Issue by ajdavis** [GetmeUK/MongoFrames#3]
Hi, I notice the guide says, "Projections also allow reference fields (fields containing an ObjectID) to be dereferenced." This implies that a field is a reference to another document if and only if it contains an ObjectId. In fact, a field can contain an ObjectId and *not* be a reference to another document. Additionally, a field could contain any other type (int, string, subdocument, ...) except array, and yet be a reference to another document.

Documents' _id fields can contain any type but array:

https://docs.mongodb.com/manual/core/document/#the-id-field

**Comment by anthonyjb** [comment]
Hi @ajdavis, I'll look to rephrase this to make it clear I'm talking about references in the context of using MongoFrames projections and not the abstract concept of referencing documents in MongoDB where as you highlight it's up to you to determine what defines a reference.

MongoFrames projections provide a short-cut for defining referenced relationships between documents so that your query returns those referenced documents but (at least for now) these relationships are limited to supporting ObjectIDs. And I agree I should make it clear this is a restriction of MongoFrames projections and is unrelated to MongoDB.

**Comment by anthonyjb** [comment]
@ajdavis I have now updated the getting started guide and API sections for projections to include the following:

> **Note:** To be clear when I talk about reference fields in MongoFrames' projections I am describing a scenario where a field's value consists of one or more **ObjectID**s that reference other documents in the same database. It is of course perfectly possible in MongoDB to reference another document using any type of data not just **ObjectID**s and for an **ObjectID** to not be a reference to another document. This is a construct of MongoFrames and not MongoDB. 

Thank you for your feedback, I'm closing this issue now but if you feel further clarification is needed please let me know.

**Comment by ajdavis** [comment]
Great, that's much better now thanks.
